### PR TITLE
Fixes #28853 - Add NIC type to safemode methods

### PR DESCRIPTION
**Pull request by lzap**
I'd appreciate quick merge, need this for IPv6 template changes.

### CODE REVIEW

File: app/models/nic/base.rb [hunk 1]
```diff
@@ -81,18 +81,34 @@ class Jail < ::Safemode::Jail
...
+        :children_mac_addresses, :nic_delay, :fqdn, :shortname, :type, :managed?, :bond?, :bridge?, :bmc?
     end
 
     # include STI inheritance column in audits
     def self.default_ignored_attributes
       super - [inheritance_column]
     end
 
+    def managed?
+      type == 'Nic::Managed'
```
Comment (ares) [inline app/models/nic/base.rb]: This is conflicting with managed boolean column. Nic::Managed class says "regular physical interface", read eth0. This would cause a regression, since `managed?` today is checked for whether we run orchestration on this NIC. Please rename to regular? or something similar

**Review by ares — CHANGES_REQUESTED** [review]

File: app/models/nic/base.rb (hunk 1 above)
Comment (lzap) [reply app/models/nic/base.rb]: Good catch, I actually only need to add this to safemode, since it exists already, I am going to remove this method and keep it in safemode allowed list.

**Comment by lzap** [comment]
Rebased, also added one additional helper which will be useful in templates: https://github.com/theforeman/community-templates/pull/706

**Comment by lzap** [comment]
Network issues with Capybara. [test foreman]

**Comment by lzap** [comment]
Rebased to fix issues.

**Comment by lzap** [comment]
Audited failures seem irrelevant.

File: app/models/host/managed.rb [hunk 2]
```diff
@@ -740,6 +740,10 @@ def pxe_loader
     explicit_pxe_loader || hostgroup.try(:pxe_loader)
   end
 
+  def pxe_loader_efi?
+    pxe_loader.include?('EFI')
```
Comment (ares) [inline app/models/host/managed.rb:744]: Is this related to this change? It seems useful, however it's not exposed in safemode, should it be added to the Host jail too?

**Review by ares — CHANGES_REQUESTED** [review]
Looks good now, one question regarding the `pxe_loader_efi?`

File: app/models/host/managed.rb (hunk 2 above)
Comment (lzap) [reply app/models/host/managed.rb:744]: It is related to the IPv6 refactoring I've done. It vastly simplifies some if-statements. It needs to be added to safemode list, I forgot. Let me do that.

**Comment by lzap** [comment]
Rebased.

**Comment by lzap** [comment]
Green @ares

**Review by ares — APPROVED** [review]
Thanks @lzap, merging!
